pkg/diskmaker/controllers/lvset/device_age.go

```diff
@@ -0,0 +1,62 @@
+package lvset
+
+import (
+	"sync"
+	"time"
+)
+
+var (
+	// deviceMinAge is the minimum age for a device to be considered safe to claim
+	// otherwise, it could be a device that some other entity has attached and we have not claimed.
+	deviceMinAge = time.Minute
+)
+
+type ageMap struct {
+	ageMap      map[string]time.Time
+	mux         sync.Mutex
+	currentTime *func() time.Time
+}
+
+func newAgeMap() *ageMap {
+	f := func() time.Time {
+		return time.Now()
+	}
+	return &ageMap{currentTime: &f}
+}
+
+// checks if older than,
+// records current time if this is the first observation of key
+func (a *ageMap) isOlderThan(key string, t time.Duration) bool {
+
+	firstObserved := a.getFirstObserved(key)
+	currentTime := *a.currentTime
+	return currentTime().Sub(firstObserved) > t
+
+}
+
+// getFirstObserved returns the age that the key was first observed and records it if
+// it was first observed now
+func (a *ageMap) getFirstObserved(key string) time.Time {
```

Comment:
`getFirstObserved` is not merely getting the value, it is also storing and updating the map. IMO - a better name is needed for this function because a casual reader of code may think that this function is readonly..